src/Evolution/DiscontinuousGalerkin/Actions/NormalVectorQuantities.hpp

```diff
@@ -0,0 +1,176 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <memory>
+#include <optional>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/Tags.hpp"
+#include "Domain/Structure/Direction.hpp"
+#include "Evolution/DiscontinuousGalerkin/NormalVectorTags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TypeTraits/CreateHasTypeAlias.hpp"
+
+namespace evolution::dg::Actions::detail {
+CREATE_HAS_TYPE_ALIAS(inverse_spatial_metric_tag)
+CREATE_HAS_TYPE_ALIAS_V(inverse_spatial_metric_tag)
+
+template <bool HasInverseSpatialMetricTag = false>
+struct inverse_spatial_metric_tag_impl {
+  template <typename System>
+  using f = tmpl::list<>;
+};
+
+template <>
+struct inverse_spatial_metric_tag_impl<true> {
+  template <typename System>
+  using f = tmpl::list<typename System::inverse_spatial_metric_tag>;
```

Comment:
Just curious...instead of SFINAE, could you have instead used if constexpr or something from TMPL.hpp to select the tmpl::list?

Reply:
No, the reason being that `if constexpr` must be used inside functions. Here we would want:
```cpp
template <bool T>
struct blah {
  if constexpr (T) {
    template <typename System>
    using f = ...;
  } else {
    template <typename System>
    using f = tmpl::list<>;
  }
};
```
C++ doesn't allow this, but other languages like D do allow this. I personally would like to be able to write code like the above, but it's unclear when such facilities will exist in C++. C++23 at the earliest, I'd be happy if it gets in by C++29 :P